src/ring_theory/roots_of_unity.lean

```diff
@@ -153,6 +153,54 @@ calc  fintype.card (roots_of_unity k R)
 ... = (nth_roots k (1 : R)).card                  : multiset.card_attach
 ... ≤ k                                           : card_nth_roots k 1
 
+variables {k R}
+
+@[norm_cast]
+lemma roots_of_unity.coe_pow (ζ : roots_of_unity k R) (m : ℕ) : ↑(ζ ^ m) = (ζ ^ m : R) :=
+begin
+  change ↑(↑(ζ ^ m) : units R) = ↑(ζ : units R) ^ m,
+  rw [subgroup.coe_pow, units.coe_pow],
+end
+
+/-- Restrict a ring homomorphism between integral domains to the nth roots of unity -/
+def ring_hom.restrict_roots_of_unity (σ : R →+* S) (n : ℕ+) :
+  roots_of_unity n R →* roots_of_unity n S :=
```

Comment:
Is this mapping unique/invertible? That is, is it possible to define
```lean
def roots_of_unity.lift_ring_hom (n : ℕ+) :
  (R →+* S) ≃ (roots_of_unity n R →* roots_of_unity n S) :=
{ to_fun := λ σ, σ.restrict_roots_of_unity,
  inv_fun := sorry }
```
If the answer is yes, can you add a `TODO` to the docstring?

Reply:
I don't think that its unique. Consider the nontrivial automorphism of Q(sqrt(2)), given by sqrt(2) -> -sqrt(2). This is a nontrivial ring homomorphism but it induces the identity on the roots of unity in Q(sqrt(2)), which are just {-1,1}.